src/standalone/Vegetation/soil_moisture_stress.jl

```diff
@@ -0,0 +1,390 @@
+- `ψ_fc`: Field capacity matric potential (m H2O), default -3.36 m (-33 kPa)
+- `ψ_wp`: Wilting point matric potential (m H2O), default -152.9 m (-1500 kPa)
+
+The function uses the `inverse_matric_potential` function to calculate the effective
+saturation at field capacity and wilting point, then converts to volumetric water content.
+"""
+function PiecewiseMoistureStressParametersFromHydrology(
+    ::Type{FT},
+    hydrology_cm,
+    ν::Union{FT, ClimaCore.Fields.Field},
+    θ_r::Union{FT, ClimaCore.Fields.Field},
+    soil_zmin::Union{FT, ClimaCore.Fields.Field};
+    c::FT = FT(1.0),
+    β0::FT = FT(1.0),
+    ψ_fc::FT = FT(-3.36734694),    # -33 kPa, equivalent in m H2O
+    ψ_wp::FT = FT(-153.06122449),  # -1500 kPa, equivalent in m H2O
+    verbose::Bool = false,
+) where {FT <: AbstractFloat}
+    if c <= 0
+        throw(
+            ArgumentError("Curvature parameter `c` must be greater than zero"),
+        )
+    end
+    if β0 <= 0
+        throw(
+            ArgumentError(
+                "Intrinsic moisture stress factor `β0` must be greater than zero",
+            ),
+        )
+    end
+
+    c = FT(c)
+    β0 = FT(β0)
+    ψ_fc = FT(ψ_fc)
+    ψ_wp = FT(ψ_wp)
+
+    # Calculate effective saturation at field capacity and wilting point
+    if hydrology_cm isa ClimaCore.Fields.Field
+        S_fc = @. ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_fc)
+        S_wp = @. ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_wp)
+        θ_high = @. S_fc * (ν - θ_r) + θ_r
+        θ_low = @. S_wp * (ν - θ_r) + θ_r
+    else
+        S_fc = ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_fc)
+        S_wp = ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_wp)
+        θ_high = S_fc * (ν - θ_r) + θ_r
+        θ_low = S_wp * (ν - θ_r) + θ_r
+    end
+
+    return PiecewiseMoistureStressParameters(
+        θ_high = θ_high,
+        θ_low = θ_low,
+        c = c,
+        β0 = β0,
+        soil_zmin = soil_zmin,
+    )
+end
+
+"""
+    compute_piecewise_moisture_stress(
+        θ_high::FT,
+        θ_low::FT,
+        c::FT,
+        β0::FT,
+        θ::FT,
+    ) where {FT}
+
+This function computes at a point the soil moisture stress factor using the volumetric water content θ (m^3/m^3)
+and four parameters: `θ_high` (field capacity, m^3/m^3), `θ_low` (wilting point, m^3/m^3), `c` (curvature parameter,
+unitless), and `β0` (intrinsic moisture stress factor, unitless). This is a modification to
+the piecewise formulation defined in Egea et al. (2011) where the entire functional form is multiplied by
+`β0` to allow for some "intrinsic moisture stress" even in well-watered conditions.
+
+Citation: https://doi.org/10.1016/j.agrformet.2011.05.019
+"""
+function compute_piecewise_moisture_stress(
+    θ_high::FT,
+    θ_low::FT,
+    c::FT,
+    β0::FT,
+    θ::FT,
+) where {FT}
+    # avoid taking e.g. sqrt of negative numbers for rational c
+    arg = max((θ - θ_low) / (θ_high - θ_low), FT(0))
```

Comment:
I think we should be OK without this, at least for the progostic case, and in the prescribed case also.

if we keep this, we do not need to clip at zero in line 335, because arg is guaranteed to be >=0 already.